src/lib/UXDController.ts

```diff
@@ -28,97 +31,107 @@ export class UXDController {
     provider,
     controllerAddress,
     uxdTokenAddress,
+    market
   }: {
     provider: providers.Provider;
     controllerAddress: string;
     uxdTokenAddress: string;
+    market: string;
   }) {
     this.provider = provider;
-
+    this.market = market;
     this.controllerContract = UXDController__factory.connect(controllerAddress, this.provider);
     this.uxdContract = ERC20__factory.connect(uxdTokenAddress, this.provider);
 
     this.registerEventListeners();
   }
 
-  public mint({
-    market,
-    collateral,
-    ethAmount,
-    slippage,
-    signer,
-  }: {
-    market: string;
-    collateral: string;
-    ethAmount: BigNumber;
-    slippage: BigNumber;
-    signer: Signer;
-  }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).mint(
-      market,
-      collateral,
-      ethAmount,
-      slippage
-    );
-  }
-
-  public redeem({
-    market,
-    collateral,
-    uxdAmount,
-    slippage,
-    signer,
-  }: {
-    market: string;
-    collateral: string;
-    uxdAmount: BigNumber;
-    slippage: BigNumber;
-    signer: Signer;
-  }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).redeem(
-      market,
-      collateral,
-      uxdAmount,
-      slippage
-    );
-  }
-
-  public mintWithEth({
-    market,
-    ethAmount,
+  public async mint({
+    amount,
     slippage,
     signer,
+    collateralToken,
   }: {
-    market: string;
-    ethAmount: BigNumber;
-    slippage: BigNumber;
+    amount: number;
+    slippage: number;
     signer: Signer;
+    collateralToken?: string;
   }): Promise<ContractTransaction> {
-    return this.controllerContract.connect(signer).mintWithEth(
-      market,
-      slippage,
-      {
-        value: ethAmount,
-      },
-    );
-  }
-
-  public redeemEth({
-    market,
-    uxdAmount,
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    const ethSlippage = ethers.utils.parseEther(slippage.toString());
+    if (collateralToken) {
+      return await this.mintWithERC20(
+        ethAmount,
+        ethSlippage,
+        signer,
+        collateralToken,
+      );
+    }
+    return await this.mintWithETH(ethAmount, ethSlippage, signer);
```

Comment:
remove `async/await`, the functions you're calling already return a `Promise`, and we don't do anything with the result that would require `await`ing the Promise.

Reply:
Oh so that's how `async/await` works 🤯